src/umockcallrecorder.c

```diff
@@ -23,25 +28,122 @@ typedef struct UMOCKCALLRECORDER_TAG
     UMOCKCALL_HANDLE* actual_calls;
     char* expected_calls_string;
     char* actual_calls_string;
+    UMOCK_C_LOCK_FACTORY_CREATE_LOCK_FUNC lock_factory_create_lock;
+    void* lock_factory_create_lock_params;
+    UMOCK_C_LOCK_HANDLE lock;
 } UMOCKCALLRECORDER;
 
-UMOCKCALLRECORDER_HANDLE umockcallrecorder_create(void)
+static void internal_lock_acquire_exclusive_if_needed(UMOCKCALLRECORDER_HANDLE call_recorder)
+{
+    if (call_recorder->lock != NULL)
+    {
+        call_recorder->lock->acquire_exclusive(call_recorder->lock);
+    }
+}
+
+static void internal_lock_acquire_shared_if_needed(UMOCKCALLRECORDER_HANDLE call_recorder)
+{
+    if (call_recorder->lock != NULL)
+    {
+        call_recorder->lock->acquire_shared(call_recorder->lock);
+    }
+}
+
+static void internal_lock_release_exclusive_if_needed(UMOCKCALLRECORDER_HANDLE call_recorder)
+{
+    if (call_recorder->lock != NULL)
+    {
+        call_recorder->lock->release_exclusive(call_recorder->lock);
+    }
+}
+
+static void internal_lock_release_shared_if_needed(UMOCKCALLRECORDER_HANDLE call_recorder)
+{
+    if (call_recorder->lock != NULL)
+    {
+        call_recorder->lock->release_shared(call_recorder->lock);
+    }
+}
+
+void internal_umockcallrecorder_reset_all_calls(UMOCKCALLRECORDER_HANDLE umock_call_recorder)
```

Comment:
>internal_umockcallrecorder_reset_all_calls [](start = 5, length = 42)

static

Reply:
Changed.

---
In reply to: [530653057](https://github.com/Azure/umock-c/pull/177#discussion_r530653057) [](ancestors = 530653057)